docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,439 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model instantiation\n",
+    "\n",
+    "The novel aspect introduced by Concret ML for KNN model is the hyperparameter `n_bits`, which represents the precision for quantizing input data. This quantization step is essential after the training phase, since FHE exclusively operates over integers."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_neighbors = 3\n",
+    "\n",
+    "concrete_knn = ConcreteKNeighborsClassifier(n_bits=3, n_neighbors=n_neighbors)\n",
+    "\n",
+    "# Fit both the Concrete ML and its equivalent float estimator on clear data\n",
+    "concrete_knn, sklearn_model = concrete_knn.fit_benchmark(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Compile the model\n",
+    "\n",
+    "\n",
+    "The compilation step aims to:\n",
+    "- convert the quantized model to its FHE equivalent\n",
+    "- create an executable operation graph\n",
+    "- check the operation graph's compatibility with FHE\n",
+    "- compute the maximum bit-width needed for model execution\n",
+    "- determine cryptographic parameters necessary for generating secret keys and evaluation keys"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Compilation time: 1.03 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "time_begin = time.time()\n",
+    "circuit = concrete_knn.compile(X)\n",
+    "print(f\"Compilation time: {time.time() - time_begin:.2f} seconds\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Maximum bit-width reached in the circuit: 8\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"Maximum bit-width reached in the circuit: {circuit.graph.maximum_integer_bit_width()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Key generation\n",
+    "\n",
+    "The circuit generated by the compiler is used to generate a set of keys:\n",
+    " - a _Secret key_ , held exclusively by the user and used for both encryption and decryption process\n",
+    "\n",
+    "- an _Evaluation Key_, publicly accessible without compromising the security of the scheme, and used to evaluate the circuit on encrypted data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Key generation time: 627.87 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# For circuits exceeding 8-bits, the key generation process might become time-consuming\n",
```

Comment:
I still believe this sentence is confusing, when I see `Key generation time: 627.87 seconds` I'm already telling myself it's time-consuming. I would at least add "much more" or change it to something like "might take up to an hour", as you told me last time !